onmt/inputters/text_dataset.py

```diff
@@ -57,6 +58,37 @@ def _feature_tokenize(
     return tokens
 
 
+class TextMultiField(RawField):
+    def __init__(self, base_name, base_field, feats_fields):
+        super(TextMultiField, self).__init__()
+        self.base_field = base_field
```

Comment:
I'm not sure it needs to have separate attributes for the base field and the feature fields. Inside the class, the distinction doesn't matter and the code in process() could be streamlined if all of the layers were in the same list, something like this:

```
levels = [f.process(b, device=device) for _, f in self.fields]
```

And then if we still want a clean way to use just the base or feature fields, we can do it with a property, i.e.

```
@property
def base_field(self):
    return self.fields[0]
```

Reply:
Good idea. I went with it. Unfortunately it doesn't streamline process() since the base field may or may not return a length whereas the feature fields never do.